### PR TITLE
Add platform detection to makefile and e2e tests

### PR DESCRIPTION
**Pull request by gkrenn**
# Description

Recently the ability to automatically detect the platform was added to our helm release.

This PR updates the makefile and the e2e tests to use this ability.
Note that the e2e tests no longer use the GitHub releases but helm for installation of specific versions.

## How can this be tested?

- `make deploy` and `make undeploy` do not require a platform.
- `make deploy` is idempotent => `helm upgrade` is used for installation to achieve that

## Checklist
- [x] Unit tests have been updated/added
- [x] PR is labeled accordingly
- [x] I have read and understood the [contribution guidelines](https://github.com/Dynatrace/dynatrace-operator/blob/main/CONTRIBUTING.md)

### CODE REVIEW

**Comment by luhi-DT** [comment]
Ci fails because of:
`test/helpers/components/operator/installation.go:74:26: printf: fmt.Sprintf format %s has arg withCsi of wrong type bool, see also https://pkg.go.dev/fmt#hdr-Printing (govet)
  		helm.WithArgs("--set", fmt.Sprintf("csidriver.enabled=%s", withCsi)),`

File: hack/make/deploy/kubernetes.mk [hunk 1]
```diff
@@ -1,24 +1,32 @@
...
+			--create-namespace \
+			--install \
+			--atomic \
 			--set installCRD=true \
-			--set platform="kubernetes" \
-			--set csidriver.enabled=true \
+			--set csidriver.enabled=$(ENABLE_CSI) \
 			--set manifests=true \
-			--set image="$(IMAGE_URI)" | kubectl apply -f -
+			--set image="$(IMAGE_URI)"
```
Comment (0sewa0) [inline hack/make/deploy/kubernetes.mk:13]: why did we keep the old `deploy/kubernetes` (and others) entrypoints ?
if we are going to keep them, shouldn't they act the same? (use helm upgrade so they are managed by helm)
also the openshift entrypoints were not updated, why? (they don't use `$(ENABLE_CSI)`)

**Review by 0sewa0 — CHANGES_REQUESTED** [review]

File: hack/make/deploy/kubernetes.mk [hunk 2]
```diff
@@ -29,26 +37,25 @@ deploy/gke-autopilot: manifests/crd/helm
 			--set manifests=true \
 			--set image="$(IMAGE_URI)" | kubectl apply -f -
 
-
+## Undeploy the operator in a cluster configured in ~/.kube/config where platform and k8s version are autodetected
+undeploy:
+	-helm uninstall dynatrace-operator \
```
Comment (0sewa0) [inline hack/make/deploy/kubernetes.mk:42]:
```suggestion
	helm uninstall dynatrace-operator \
```

File: hack/make/deploy/kubernetes.mk (hunk 1 above)
Comment (gkrenn) [reply hack/make/deploy/kubernetes.mk:13]: I kept the entries on purpose to give us the possibility to still deploy/undeploy the operator without helm from the makefile.

Will rename `deploy` to `deploy/helm` and move it in it's own file. 

Concerning the openshift entries, I will adapt them too

File: hack/make/deploy/kubernetes.mk (hunk 2 above)
Comment (gkrenn) [reply hack/make/deploy/kubernetes.mk]: "-" would swallow the error. We agreed on removing it.

**Review by 0sewa0 — APPROVED** [review]

**Pull request merged**